### DISCUSSION TITLE
How to reset process?

### DISCUSSION THREAD

**Discussion by gvinson** [eighty9nine/filament-approvals#34] [Q&A]
We have a publish/unpublish button once the model is approved. If a model is "unpublished", we would like to restart the entire approval flow. How can we do that? I've tried using the `undoLastApproval()` method as well as the `return()` method being called a few times. Any ideas?

**Comment by gvinson** [comment]
I've been resetting it like so but it feels clunky and doesn't work well with the table row widget
```
$items = [];
foreach (collect($page->approvalStatus->steps ?? []) as $item) {
    $item['process_approval_action'] = null;
    $item['process_approval_id'] = null;
    $items[] = $item;
}
$page->approvalStatus->steps = $items;
$page->approvalStatus->status = 'Created';
```

**Comment by gvinson** [comment]
@eighty9nine have you ran into this?

**Comment by eighty9nine** [comment]
Currently that feature is not supported in the underlying package https://github.com/ringlesoft/laravel-process-approval. We are looking into it
